website/integrations: Semaphore

nicedevil007 opened the pull request:
## Details

Created the documentation for Semaphore UI.

---

## Checklist

-   [X] Local tests pass (`ak test authentik/`)
-   [X] The code has been formatted (`make lint-fix`)

If changes to the frontend have been made

-   [X] The code has been formatted (`make web`)

If applicable

-   [X] The documentation has been updated
-   [X] The documentation has been formatted (`make website`)

nicedevil007 committed:
Initial start of Semaphore UI docs

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Added Semaphore UI docs link

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Changed to mdx format

nicedevil007 committed:
Ran make lint-fix, make web and make website

nicedevil007 committed:
Added sidebar entry

nicedevil007 committed:
Merge branch 'semaphore' of https://github.com/nicedevil007/authentik into semaphore

nicedevil007 committed:
added sidebar integration

nicedevil007 committed:
now fixed sidebar integration

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

tanberry commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+> Semaphore UI is a modern web interface for managing popular DevOps tools.
+> -- https://semaphoreui.com/
+>
+> This guide explains how to set up Semaphore UI to use authentik as the OAuth provider for the login to the WebGUI.
```
```suggestion
> This guide explains how to configure Semaphore UI to use authentik as the OAuth provider for logging in to the Web GUI.
```

nicedevil007 replied:
oh I used it from my previous PR's that I made here :D (take a look @ engomo f.e.).

tanberry commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+
+## Preparation
+
+The following placeholders will be used:
```
```suggestion
The following placeholders are used in this guide:
```

tanberry commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+
+## Semaphore UI configuration
+
+Create an application and an OAuth2/OpenID provider in authentik. Use the following parameters for the OAuth2/OpenID provider:
```
```suggestion
[Create](https://docs.goauthentik.io/docs/add-secure-apps/applications/manage_apps#add-new-applications) an application and an OAuth2/OpenID provider in authentik. Use the following parameters for the OAuth2/OpenID provider:
```

tanberry commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+## Semaphore UI configuration
+
+Create an application and an OAuth2/OpenID provider in authentik. Use the following parameters for the OAuth2/OpenID provider:
+
```
Please switch the order of these sections, so that the application part comes first, then the provider, to reflect the order of the Wizard process.

tanberry commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+- `authentik.company` is the FQDN of the authentik install.
+- `ak.cert` is the self-signed certificate that will be used for the service provider.
+
+## Semaphore UI configuration
```
Did you mean to title this "authentik configuration"?

tanberry replied:
HI @nicedevil007 help me understand this one.. don't we want line 24 to be "authentik configuration", because this section is about creating the app and provider in authentik. Then line 45 is the Semaphore section, no?

tanberry commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+
+## Semaphore UI configuration
+
+Login to your Semaphore UI host via SSH. Edit the `config.json` (should be located under `/etc/semaphore`) file with the texteditor of your choice.
```
```suggestion
Log in to your Semaphore UI host via SSH. Edit the `config.json` file (should be located under `/etc/semaphore`) file with the text editor of your choice.
```

tanberry commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+## Semaphore UI configuration
+
+Login to your Semaphore UI host via SSH. Edit the `config.json` (should be located under `/etc/semaphore`) file with the texteditor of your choice.
+For example just use this command:
```
```suggestion
For example, with the nano editor you can use this command:
```

tanberry commented on website/integrations/services/semaphore/index.mdx:
````diff
@@ -0,0 +1,87 @@
+nano /etc/semaphore/config.json
+```
+
+Before the last curly brace, add this part.
````
```suggestion
Before the last curly brace, add the following content:
```

tanberry commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+
+:::note It is mandatory to include 'authentik' in lowercase letters. There should also be another curly brace above these lines. Make sure to add a `,` after it to maintain proper formatting. :::
+
+More information on this can be found in the semaphore documentation https://docs.semaphoreui.com/administration-guide/openid/authentik/.
```
```suggestion
More information on this can be found in the Semaphore documentation https://docs.semaphoreui.com/administration-guide/openid/authentik/.
```

tanberry commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+
+Leave the rest as default.
+
+## Test the login
```
Thank you for adding this section! I love a good verification section. ;-)

nicedevil007 replied:
I learned that from you with my "old" engomo PR ;)

tanberry reviewed (changes requested):
Thanks so much @nicedevil007 for adding this integration guide! A few requested edits, please, then let's merge!

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: 4d62 <github-user@sdko.org>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: 4d62 <github-user@sdko.org>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 committed:
Update website/integrations/services/semaphore/index.mdx

Co-authored-by: 4d62 <github-user@sdko.org>
Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 commented on website/integrations/services/semaphore/index.mdx:
```diff
@@ -0,0 +1,87 @@
+- Client type: `Confidential`
+- Redirect URIs/Origins (RegEx): `https://semaphore.company/api/auth/oidc/authentik/redirect/`
+- Signing Key: `ak.cert`
+- Scopes: `authentik default OAuth Mapping: OpenID 'email', OpenID 'openid'` and `OpenID 'profile'`
```
I like to do it this way so no question leaves open. But I can remove it ofc :)

nicedevil007 committed:
Update to Feedback on PR ;)

Signed-off-by: NiceDevil <17103076+nicedevil007@users.noreply.github.com>

nicedevil007 commented:
thank you again for helping me out with my doc PR :) I hope the changes I made make more sense now.

tanberry commented:
Hi @nicedevil007 I accidentally added a comment to your comit instead of this PR... we still need to change the first (and duplicate) title "Semaphore configuration" (line 24?) to say "authentik configuration". Otherwise, looks good!

nicedevil007 committed:
Title…

Ahhh now I got what you were trying to Tell me on the last comment 😂 happy new year ✌️

tanberry reviewed (approved):
Thanks so much @nicedevil007 !
